src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -77,24 +65,25 @@ public RepoConfiguration(String location, String branch) throws InvalidLocationE
      */
     public RepoConfiguration(String location, String branch, List<String> formats, List<String> ignoreGlobList,
             boolean isStandaloneConfigIgnored, List<String> ignoreCommitList) throws InvalidLocationException {
-        this.location = location;
+        this.location = new Location(location);
         this.branch = branch;
         this.ignoreGlobList = ignoreGlobList;
         this.isStandaloneConfigIgnored = isStandaloneConfigIgnored;
-        this.formats = formats;
 
-        validateIgnoreCommits(ignoreCommitList);
-        this.ignoreCommitList = ignoreCommitList;
+        FormatList.validateFormats(formats);
+        this.formats.setFormats(formats);
```

Comment:
actually... on a second look, why isn't this using a constructor?

Reply:
Good point